### PR TITLE
control-service: add amazon ses smtp mail notifications

### PR DESCRIPTION
**Pull request by murphp15**
# Why
We want to be able to send email notifications using Amazon SES SMTP. 

# What
Until now, we only had default SMTP configuration for notifications. We move this logic to a new DefaultEmailNotification class which implements the EmailNotification interface. We also introduce an Amazon-oriented AmazonSesEmailNotification class which also implements the EmailNotification interface.

# How was this tested
It was just SMTP code until now. 
It was tested in an internal k8s cluster as we don't have a present testing SES.

Signed-off-by: Yoan Salambashev <yoan.salambashev@broadcom.com>

### CODE REVIEW

**Review by yonitoo — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/AmazonSesEmailNotification.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                    .withRegion(this.region)
+                    .build())
+            .build();
+
+    var sesClient =
+        AmazonSimpleEmailServiceClientBuilder.standard()
+            .withCredentials(new AWSStaticCredentialsProvider(credentialsProvider.getCredentials()))
+            .withRegion(this.region)
+            .build();
+    log.info("Send AmazonEmailNotification");
```
Comment (murphp15) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/AmazonSesEmailNotification.java:60]: can you change this to debug.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/DefaultEmailNotification.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
+/*
```
Comment (murphp15) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/DefaultEmailNotification.java:1]: This is not new code. We renamed this class from EmailNotification.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/EmailNotification.java [hunk 3]
```diff
@@ -5,18 +5,7 @@
```
Comment (murphp15) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/EmailNotification.java:5]: This is actually a new interface. But the interface just so happens to have the name of hte previous class.

**Review by duyguHsnHsn — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/DefaultEmailNotification.java (hunk 2 above)
Comment (duyguHsnHsn) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/DefaultEmailNotification.java:1]: Add this a comment as well

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/EmailNotification.java (hunk 3 above)
Comment (murphp15) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/notification/EmailNotification.java:5]: what do you mean add as a comment to the code. 
After this is merged why would ppl want to know that?

**Pull request merged**